lib/ts/recipe/session/with-jwt/recipeImplementation.ts

```diff
@@ -131,72 +188,46 @@ export default function (
-                    userContext,
-                });
-            }
-
-            let existingJwt = accessTokenPayload[existingJwtPropertyName];
-            assert.notStrictEqual(existingJwt, undefined);
-
-            let currentTimeInSeconds = Date.now() / 1000;
-            let decodedPayload = JsonWebToken.decode(existingJwt, { json: true });
+        mergeIntoAccessTokenPayload: async function ({
+            sessionHandle,
+            accessTokenPayloadUpdate,
+            userContext,
+        }: {
+            sessionHandle: string;
+            accessTokenPayloadUpdate: JSONObject;
+            userContext: any;
+        }): Promise<boolean> {
+            let sessionInformation = await this.getSessionInformation({ sessionHandle, userContext });
 
-            // JsonWebToken.decode possibly returns null
-            if (decodedPayload === null) {
-                throw new Error("Error reading JWT from session");
+            let newAccessTokenPayload = { ...sessionInformation.accessTokenPayload, ...accessTokenPayloadUpdate };
+            for (const key of Object.keys(accessTokenPayloadUpdate)) {
+                if (accessTokenPayloadUpdate[key] === null) {
+                    delete newAccessTokenPayload[key];
+                }
             }
 
-            let jwtExpiry = decodedPayload.exp - currentTimeInSeconds;
+            return jwtAwareUpdateAccessTokenPayload(sessionInformation, newAccessTokenPayload, userContext);
+        },
 
-            if (jwtExpiry <= 0) {
-                // it can come here if someone calls this function well after
-                // the access token and the jwt payload have expired. In this case,
-                // we still want the jwt payload to update, but the resulting JWT should
-                // not be alive for too long (since it's expired already). So we set it to
-                // 1 second lifetime.
-                jwtExpiry = 1;
-            }
+        /**
+         * @deprecated use mergeIntoAccessTokenPayload instead
+         */
+        updateAccessTokenPayload: async function ({
+            sessionHandle,
+            newAccessTokenPayload,
+            userContext,
+        }: {
+            sessionHandle: string;
+            newAccessTokenPayload: any;
+            userContext: any;
+        }): Promise<boolean> {
+            newAccessTokenPayload =
+                newAccessTokenPayload === null || newAccessTokenPayload === undefined ? {} : newAccessTokenPayload;
 
-            newAccessTokenPayload = await addJWTToAccessTokenPayload({
-                accessTokenPayload: newAccessTokenPayload,
-                jwtExpiry,
-                userId: sessionInformation.userId,
-                jwtPropertyName: existingJwtPropertyName,
-                openIdRecipeImplementation,
-                userContext,
-            });
+            const sessionInformation = await this.getSessionInformation({ sessionHandle, userContext });
```

Comment:
This can return undefined as well.. need to check the type and return false if this is undefined. You should also add `this` to the first param of this function (and others) so that TS complains about this.